pkg/sql/create_table.go

```diff
@@ -2409,6 +2375,17 @@ func newTableDesc(
 		)
 	})
 
+	// We need to ensure sequence ownerships so that column owned sequences are
+	// correctly dropped when a column/table is dropped.
+	for colName, seqDesc := range colNameToOwnedSeq {
```

Comment:
i think this is broken if there is an error during `NewTableDesc`. in particular, you need an `if err != nil { return ret, err}` check above this stanza

Reply:
```
root@127.0.0.1:26257/defaultdb> SET serial_normalization = sql_sequence;
SET


Time: 0ms total (execution 0ms / network 0ms)

root@127.0.0.1:26257/defaultdb>
CREATE TABLE test_serial (
        a INT PRIMARY KEY,
        b SERIAL
) with (fillfactor = 70000);

*
* ERROR: a SQL panic has occurred while executing the following statement:
* CREATE TABLE test_serial (a INT8 PRIMARY KEY, b INT8 NOT NULL DEFAULT nextval('defaultdb.public.test_serial_b_seq')) WITH (fillfactor = 70000)
*
*
* ERROR: a panic has occurred!
* runtime error: invalid memory address or nil pointer dereference
* (1) attached stack trace
*   -- stack trace:
*   | github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:678
*   | [...repeated from below...]
* Wraps: (2) while executing: CREATE TABLE _ (_ INT8 PRIMARY KEY, _ INT8 NOT NULL DEFAULT nextval('_')) WITH (_ = _)
* Wraps: (3) attached stack trace
*   -- stack trace:
*   | github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:678
*   | runtime.gopanic
*   | 	/usr/local/opt/go/libexec/src/runtime/panic.go:1038
*   | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:58
*   | runtime.gopanic
*   | 	/usr/local/opt/go/libexec/src/runtime/panic.go:1038
*   | runtime.panicmem
*   | 	/usr/local/opt/go/libexec/src/runtime/panic.go:221
*   | runtime.sigpanic
*   | 	/usr/local/opt/go/libexec/src/runtime/signal_unix.go:735
*   | github.com/cockroachdb/cockroach/pkg/sql.setSequenceOwner
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/create_table.go:2606
*   | github.com/cockroachdb/cockroach/pkg/sql.newTableDesc
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/create_table.go:2248
*   | github.com/cockroachdb/cockroach/pkg/sql.(*createTableNode).startExec
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/create_table.go:375
*   | github.com/cockroachdb/cockroach/pkg/sql.startExec.func2
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:516
*   | github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal.func1
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:112
*   | github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visitInternal
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:295
*   | github.com/cockroachdb/cockroach/pkg/sql.(*planVisitor).visit
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:79
*   | github.com/cockroachdb/cockroach/pkg/sql.walkPlan
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/walk.go:43
*   | github.com/cockroachdb/cockroach/pkg/sql.startExec
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan.go:519
*   | github.com/cockroachdb/cockroach/pkg/sql.(*planNodeToRowSource).Start
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/plan_node_to_row_source.go:146
*   | github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Columnarizer).Init
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/columnarizer.go:158
*   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*batchInfoCollector).Init
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/stats.go:87
*   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*BatchFlowCoordinator).init.func1
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:235
*   | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:91
*   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*BatchFlowCoordinator).init
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:234
*   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*BatchFlowCoordinator).Run
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/flow_coordinator.go:279
*   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Run
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:259
*   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:573
*   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:1409
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1411
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:1088
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:718
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:137
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func1
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1742
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1746
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run
*   | 	/Users/otan/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1675
* Wraps: (4) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) *safedetails.withSafeDetails (3) *withstack.withStack (4) runtime.errorString
* ```

Reply:
you'll want #75841 too